monad-archive/src/test_utils.rs

```diff
@@ -80,14 +80,35 @@ pub fn mock_block(number: u64, transactions: Vec<TxEnvelopeWithSender>) -> Block
     }
 }
 
+pub struct TestMinioContainer {
+    pub port: u16,
+}
+
+impl TestMinioContainer {
+    pub async fn new() -> Result<Self> {
+        let port = NEXT_PORT.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        Command::new("docker")
+            .arg("run")
+            .arg("-p")
+            .arg(format!("{port}:9000"))
+            .arg("minio/minio")
+            .arg("server")
+            .arg("/tmp/minio")
+            .spawn()
```

Comment:
docker subprocess dies once test process dies

Reply:
I was wrong here, reverted to killing the container in a custom `Drop` impl like I did for mongo